### ISSUE TITLE
No Devices to cast to - Bridge has TCP connection

### ISSUE THREAD

**Issue by xWTyrone** [hensm/fx_cast#81]
Version: 0.0.4
Bridge Version: 0.0.4
Firefox Version: 69.0 64-bit
OS: Windows 10 Pro x64 (Build 18362.295)

Target: Chromecast Audio
FW: 1.42.168007 (from Google Home app)

Happens when casting from extension icon, right-click menu and website app (tried Spotify and Soundcloud).

I have no clue where logs are stored for either the extension or the bridge.

Whenevery I try to cast, there is no device available. Google Home app reports otherwise.
Using `netstat`, I can see the bridge successfully connected to the CCA - port 8009.

**Comment by hensm** [comment]
Does it work on v0.0.3?

What do you see when you try to cast? Does the cast button appear? If you're clicking a menu item, does it just do nothing?

There are some debug builds in the latest comment in #79, since it sounds like it could be the same problem.

**Comment by xWTyrone** [comment]
> Does it work on v0.0.3?

v0.0.3 works. The option to cast shows up. Tested with Twitch.

> What do you see when you try to cast?

The cast window. With the dropdown to select what to cast, but no cast option.

> Does the cast button appear?

Which button? The one in the cast window? No. The context menu (right-click)? Yes. Extension Toolbar? Yes.

> If you're clicking a menu item, does it just do nothing?

Which menu?

> There are some debug builds in the latest comment in #79 

Will check later.

I'll collect some images and link them here for clarity's sake. Then I'll check debug builds.

**Comment by hensm** [comment]
> v0.0.3 works

Okay, definitely a regression then. Pretty sure I know which commit caused it. It's a squashed PR that got way too big. Will work on this when I have the time.

> Which menu?`

Extension "Cast..." menu item.

**Comment by shmerl** [comment]
I think have the same bug (tested latest release 0.0.5). No devices to cast to show up, while VLC for example finds my BenQ GV1 projector.

**Comment by hensm** [comment]
@shmerl Could be. Logs would help:
https://extensionworkshop.com/documentation/develop/debugging/

**Comment by edoren** [comment]
I'm having this same issue with a Chromecast 3 in v0.0.6, Firefox 73.0.1, this are the debug logs for the extension:

```
fx_cast (Info): init background.js:1:1185
fx_cast (Info): init (menus) background.js:1:1185
fx_cast (Info): init (request listener) background.js:1:1185
fx_cast (Info): init (whitelist) background.js:1:1185
fx_cast (Info): init (media overlay) background.js:1:1185
fx_cast (Info): init (browser action) background.js:1:1185
fx_cast_bridge disconnected: <empty string> background.js:1:11071
fx_cast (Error): Bridge connection failed and backup not enabled. background.js:1:1279
fx_cast_bridge disconnected: <empty string> background.js:1:11071
fx_cast (Error): Bridge connection failed and backup not enabled. background.js:1:1279
fx_cast (Info): Cancelled receiver selection background.js:1:1185
fx_cast_bridge disconnected: <empty string> background.js:1:11071
fx_cast (Error): Bridge connection failed and backup not enabled. background.js:1:1279
fx_cast_bridge disconnected: <empty string> background.js:1:11071
```

For some reason the bridge keeps disconnecting, was trying to check the message connection between the extension and the bridge but it's hard to debug it not knowing the message protocol used.

**Comment by DanielRios549** [comment]
I don't know if this is the same bug, but for me, there is nothing to cast to as well, I use KDE Neon 5.18.5 and Firefox 76.0.1 (the one from Ubuntu 18.04 repository). I installed Chrome to make sure it is working, it shows my old smart TV and the Chromecast I named "Chromecast", but Firefox does not.

On Chrome:
![image](https://user-images.githubusercontent.com/11152535/82960544-2f3fb500-9f91-11ea-9aee-130ea9723c46.png)

On Firefox with both extension and bridge installed:
![image](https://user-images.githubusercontent.com/11152535/82960635-66ae6180-9f91-11ea-8906-0cdbf4eefecc.png)

**Comment by hensm** [comment]
@DanielRios549 Can you try this build? https://github.com/hensm/fx_cast/issues/113#issuecomment-624287370

**Comment by DanielRios549** [comment]
@hensm Still not working, it show nothing.

**Comment by ugurbor** [comment]
@hensm I am having the same issue with @DanielRios549. Latest Ubuntu 20.04. Can see the TV in Chromium but can't in fx_cast.

![image](https://user-images.githubusercontent.com/3953431/83353970-d55b3880-a35e-11ea-8e94-76cdcbcc49c1.png)

**Comment by hensm** [comment]
@ugurbor I don't think that's a Chromecast receiver, especially since Avahi isn't picking it up.

**Comment by ugurbor** [comment]
@hensm yeah it's not a Chromecast device but it's a Google Cast device. I thought fx_cast also supports that. Maybe I should search with a different protocol/port with avahi? Since Chrome can find it, it can be supported I guess?

**Comment by hensm** [comment]
@ugurbor I don’t think it’s Google Cast, maybe DIAL? Does it work with other apps than YouTube on Chrome?

**Comment by ugurbor** [comment]
@hensm it also works with the YouTube apps on mobile devices. Also Netflix and I believe Spotify works with apps.

**Comment by DanielRios549** [comment]
@ugurbor Hum, my Chromecast is not shown in avahi as well, I change from KDE Neon to PopOS and the problem persists. But Chrome still can access it and even my old Samsung Smart TV

**Comment by Ingineerix** [comment]
Also "No receiver devices found" on Ubuntu 20.04 with 0.0.6 in Firefox 77.0.1

Both Chromecasts do show up on avahi:
Found service 'Chromecast-a6624405a8b11dbe31114e96f4bd7f53' of type '_googlecast._tcp' in domain 'local' on 3.0.
Found service 'Chromecast-21aa4201ffa9c979c7af0c07fb960b6c' of type '_googlecast._tcp' in domain 'local' on 3.0.

**Comment by hensm** [comment]
@Ingineerix 
See #121. Just testing now, and fixing some Linux issues, but should have a new release in the next week or so.

**Comment by outlandnish** [comment]
I have the same issues as @edoren on Windows 10 w/ Firefox. The bridge keeps disconnecting with no reason (empty string). I ran a local build of it and had the same result in daemon mode too.

**Comment by hensm** [comment]
@outlandnish Likely the bridge is sending invalid output to stdout and Firefox is closing it. I've seen that error before, but I don't remember the exact circumstances. Filed a new issue (#134).
